src/Workspaces/Core/Portable/Storage/DesktopPersistenceStorageServiceFactory.cs

```diff
@@ -69,6 +69,16 @@ public IWorkspaceService CreateService(HostWorkspaceServices workspaceServices)
             return NoOpPersistentStorageService.GetOrThrow(options);
         }
 
+        private static StorageDatabase GetDatabase(HostWorkspaceServices workspaceServices)
+        {
+            var experimentationService = workspaceServices.GetService<IExperimentationService>();
+            if (experimentationService?.IsExperimentEnabled(WellKnownExperimentNames.CloudCache) == true)
+                return StorageDatabase.CloudCache;
+
+            var optionService = workspaceServices.GetRequiredService<IOptionService>();
+            return optionService.GetOption(StorageOptions.Database);
```

Comment:
❗ `StorageOptions.Database` needs a `LocalUserProfileStorageLocation` so the experimental path can be manually set.

Reply:
Take a look at the definition of `StorageOptions.SQLiteInMemoryWriteCache` prior to 44ea1444834d7563ffb1e2a30911f355a3fa8a06 for an example.

https://github.com/CyrusNajmabadi/roslyn/blob/050d32d8eeabe34653f8bf35da0fa07bb06eb067/src/Workspaces/Core/Portable/Storage/StorageOptions.cs#L23-L25

Reply:
can do.